src/move/sources/emojicoin_dot_fun.move

```diff
@@ -2437,8 +2439,52 @@ module emoji::main {
     const ZOMBIE: vector<u8> = x"f09fa79f";
     const ZZZ: vector<u8> = x"f09f92a4";
 
-    fun init_module(_: &signer) {
+    #[resource_group = ObjectGroup]
+    struct Market has key {
+        market_id: address,
+        market_address: address,
+        emoji_bytes: vector<u8>,
+        extend_ref: ExtendRef,
+    }
+
+    #[resource_group = ObjectGroup]
+    struct Registry has key {
+        registry_address: address,
+        supported_emojis: SmartTable<vector<u8>, u8>,
+        markets_by_emoji_bytes: SmartTable<vector<u8>, address>,
```

Comment:
These are supposed to be the same keys right? Any reason not to use the market address as the value in `supported_emojis` instead?

Reply:
Where in the case it's not set yet it'd be `0x0` or something.? Or an `Option<address>` even

But I suppose it doesn't matter much, just duplicate keys

Reply:
You can create a market from an emoji combo, so this would only apply if we enforced a single emoji per market

Reply:
Oh yes of course...